include/oneapi/mkl/dft/detail/commit_impl.hpp

```diff
@@ -26,42 +26,88 @@
 #include <CL/sycl.hpp>
 #endif
 
+#include "descriptor_impl.hpp"
+
 namespace oneapi::mkl {
 enum class backend;
 }
 
 namespace oneapi::mkl::dft::detail {
 
-enum class precision;
-enum class domain;
 template <precision prec, domain dom>
 class dft_values;
 
 template <precision prec, domain dom>
 class commit_impl {
+    sycl::queue queue_;
+    mkl::backend backend_;
+
 public:
+    using descriptor_type = typename oneapi::mkl::dft::detail::descriptor<prec, dom>;
+    using fwd_type = typename descriptor_info<descriptor_type>::forward_type;
+    using bwd_type = typename descriptor_info<descriptor_type>::backward_type;
+    using scalar_type = typename descriptor_info<descriptor_type>::scalar_type;
+
     commit_impl(sycl::queue queue, mkl::backend backend) : queue_(queue), backend_(backend) {}
 
     // rule of three
-    commit_impl(const commit_impl& other) = delete;
-    commit_impl& operator=(const commit_impl& other) = delete;
+    commit_impl(const commit_impl &other) = delete;
+    commit_impl &operator=(const commit_impl &other) = delete;
     virtual ~commit_impl() = default;
 
-    sycl::queue& get_queue() noexcept {
+    sycl::queue &get_queue() noexcept {
         return queue_;
     }
 
     mkl::backend get_backend() const noexcept {
         return backend_;
     }
 
-    virtual void* get_handle() noexcept = 0;
+    virtual void *get_handle() noexcept = 0;
 
-    virtual void commit(const dft_values<prec, dom>&) = 0;
+    virtual void commit(const dft_values<prec, dom> &) = 0;
 
-private:
-    sycl::queue queue_;
-    mkl::backend backend_;
+    virtual void forward_ip_cc(descriptor_type &desc, sycl::buffer<fwd_type, 1> &inout) = 0;
```

Comment:
I don't see where these new functions are overrided for the derived commit impls. What am I missing?

Reply:
src/dft/backends/backend_compute_signature.cxx